### PR TITLE
web: drop `.theme-redesign` class

### PR DESCRIPTION
**Pull request by valerybugakov**
## Changes

- Removed the `.theme-redesign` class from the Web app and Storybook.

Closes https://github.com/sourcegraph/sourcegraph/issues/20847. 🚀 🥳

### CODE REVIEW

File: client/branded/src/components/Toggle.scss [hunk 1]
```diff
@@ -2,25 +2,14 @@ $toggle-width: 2rem;
...
+    --toggle-bar-bg: var(--icon-color);
     --toggle-bar-bg-on: var(--primary);
-    --toggle-knob-bg: var(--text-muted);
-    --toggle-knob-bg-on: var(--primary);
-    --toggle-bar-opacity: 0.2;
-    --toggle-bar-focus-opacity: 0.5;
-    --toggle-knob-disabled-opacity: 0.2;
-    --toggle-bar-focus-box-shadow: #{$box-shadow-spacing}, 0 0 0 0.1875rem var(--primary);
-
-    .theme-redesign & {
```
Comment (valerybugakov) [inline client/branded/src/components/Toggle.scss:14]: Somehow we missed it in our tracking spreadsheet, but it's a simple cleanup.

File: client/web/src/Layout.tsx [hunk 2]
```diff
@@ -279,15 +272,7 @@ export const Layout: React.FunctionComponent<LayoutProps> = props => {
                         !isSearchConsolePage &&
                         !isSearchNotebookPage
                     }
-                    variant={
-                        hideGlobalSearchInput
-                            ? 'no-search-input'
```
Comment (valerybugakov) [inline client/web/src/Layout.tsx:284]: Based on [this comment](https://github.com/sourcegraph/sourcegraph/pull/24194/files#diff-ea755d7329ab355145cfd07154985058101aaf40ef166ea10971efdca3fc857bL100) we don't need the `no-search-input` variant anymore.

File: client/web/src/integration/utils.ts [hunk 3]
```diff
@@ -87,8 +87,7 @@ export interface PercySnapshotConfig {
 }
 
 /**
- * Takes a Percy snapshot in 4 variants:
```
Comment (vovakulikov) [inline client/web/src/integration/utils.ts:90]: 💪

**Review by vovakulikov — APPROVED** [review]
🆒

**Review by umpox — APPROVED** [review]
Milestone!

**Pull request merged**